File: server/Models/GreenhouseHarvestChemical.cs
namespace Server.Models;

/// <summary>
/// A chemical applied to a <see cref="GreenhouseHarvest"/>, with the date it was applied and what
/// it cost. The cost is rolled into the harvest's final expenses. Plain CRUD — applying a chemical
/// moves no stock balance. The greenhouse counterpart to <see cref="HarvestChemical"/>.
/// </summary>
public class GreenhouseHarvestChemical
{
    public int Id { get; set; }
    public int GreenhouseHarvestId { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateOnly Date { get; set; }
    public decimal Cost { get; set; }
}

File: web/src/types/greenhouse-harvest-chemical.ts
/** A chemical applied to a greenhouse harvest, with its cost — folded into the harvest's final
 * expenses. */
export type GreenhouseHarvestChemical = {
  id: number;
  greenhouseHarvestId: number;
  name: string;
  /** ISO date string (YYYY-MM-DD). */
  date: string;
  cost: number;
};

export type GreenhouseHarvestChemicalInput = Omit<GreenhouseHarvestChemical, 'id'>;
